src/envelope.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::errors::{InternalPakeError, PakeError};
+use generic_array::{
+    typenum::{Unsigned, U32},
+    GenericArray,
+};
+use hkdf::Hkdf;
+use hmac::{Hmac, Mac, NewMac};
+use rand_core::{CryptoRng, RngCore};
+use sha2::{Digest, Sha256};
+
+// Constant string used as salt for HKDF computation
+const STR_ENVU: &[u8] = b"EnvU";
+
+/// The length of the "export key" output by the client registration
+/// and login finish steps
+pub(crate) type ExportKeySize = <Sha256 as Digest>::OutputSize;
```

Comment:
(for my edification:) Do we plan the digest to change, here? In particular I'm wondering w.r.t #15 if there are areas where a change in digests makes more or less sense than others.

Reply:
Yes, the digest here could change. I don't think we should hardcode Sha256 anywhere.